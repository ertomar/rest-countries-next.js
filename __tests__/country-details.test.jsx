import { render, screen } from '@testing-library/react';
import CountryDetails from '../src/pages/country/[id]/index';
import '@testing-library/jest-dom';

import * as nextRouter from 'next/router';

nextRouter.useRouter = jest.fn();

nextRouter.useRouter.mockImplementation(() => ({
  route: '/',
  push: jest.fn(),
}));

describe('Country Details', () => {
  it('renders CountryDetails unchanged', () => {
    const { container } = render(<CountryDetails />);
    expect(container).toMatchSnapshot();
  });

  it('renders a heading', () => {
    render(<CountryDetails />);

    const heading = screen.getByRole('heading', {
      name: /Countries App/i,
    });

    expect(heading).toBeInTheDocument();
  });

  it('renders a back button', () => {
    render(<CountryDetails />);

    const button = screen.getByRole('button', {
      name: /Back/i,
    });

    expect(button).toBeInTheDocument();
  });

  it('back button redirects to home page', () => {
    render(<CountryDetails />);

    const button = screen.getByRole('button', {
      name: /back/i,
    });

    button.click();

    expect(nextRouter.useRouter).toHaveBeenCalled();
  });
});
