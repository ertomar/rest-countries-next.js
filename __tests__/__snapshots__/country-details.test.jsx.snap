// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Details renders CountryDetails unchanged 1`] = `
<div>
  <main
    class="main-layout"
  >
    <h1
      class="headline"
    >
      Countries App
    </h1>
    <button
      class="button"
      style="color: white;"
    >
      &lt; back
    </button>
    <section
      class="CountryDetailsContainer"
    >
      <div
        class="imageContainer"
      >
        <img
          alt="Flag image"
          class="flagImage"
          data-nimg="1"
          decoding="async"
          height="100"
          src=""
          style="color: transparent;"
          width="100"
        />
      </div>
      <div
        class="infoCard"
      >
        <h3
          style="color: rgb(0, 0, 0);"
        />
        <h3
          style="color: rgb(0, 0, 0);"
        >
          Capital: 
          No Capital Provided
        </h3>
        <h3
          style="color: rgb(0, 0, 0);"
        >
          Population: 
        </h3>
        <h3
          style="color: rgb(0, 0, 0);"
        >
          Currency: 
          No Currency Provided
        </h3>
        <h3
          style="color: rgb(0, 0, 0);"
        >
          Languages: 
        </h3>
      </div>
      <div
        class="borderingCountries"
      >
        <h2
          style="color: rgb(0, 0, 0);"
        >
          Bordering Countries
        </h2>
        <div
          class="borderingEmptyState"
        >
          <p
            style="color: rgb(0, 0, 0);"
          >
            No bordering countries
          </p>
        </div>
      </div>
    </section>
  </main>
</div>
`;
